pkg/cloud/services/networking/service.go

```diff
@@ -89,15 +89,25 @@ func (s *Service) replaceAllAttributesTags(eventObject runtime.Object, resourceT
 		record.Warnf(eventObject, "FailedReplaceAllAttributesTags", "Invalid resourceType argument in function call")
 		panic(fmt.Errorf("invalid argument: resourceType, %s, does not match allowed arguments: %s or %s", resourceType, trunkResource, portResource))
 	}
+	// remove duplicate values from tags
```

Comment:
as follow up ,we can create some ut (to be honest we don't have too much so not a mandatory thing for this PR )
to test this works ..

Reply:
This one is more of a user's mistake that gives duplicate tags, not a use case. 
So, we may not need a test case for it. 
However, the previous test was run to validate the code: see https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1112#issuecomment-1012206109

Reply:
you are right :)
thanks for the great info :~